.circleci/workflows.yml

```diff
@@ -1009,27 +1481,33 @@ workflows:
                 var-name: AZURE_APPLICATION_SECRET
 
   sonarcloud_analysis_back:
+    when:
+      and:
+        - equal: [ pull_requests, << pipeline.parameters.gio_action >> ]
     jobs:
-      - keeper:
+      - setup:
           context:
             - cicd-orchestrator
       - coverage_build:
           requires:
-            - keeper
+            - setup
       - run_sonarcloud_analysis:
           context:
             - cicd-orchestrator
           requires:
             - coverage_build
 
   sonarcloud_analysis_front:
+    when:
+      and:
+        - equal: [ pull_requests, << pipeline.parameters.gio_action >> ]
```

Comment:
I would say "gio_action not in release, publish_rpms, publish_docker_images".
Sonar need to be executed during the PR to have access to quality gates but also after the merge in oder to have the complete analyse on the main branch